task_assign/recharge[] vector